src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirerFactory.cs

```diff
@@ -157,7 +179,56 @@ public ITokenAcquirer GetTokenAcquirer(string authority, string region, string c
         /// <exception cref="NotImplementedException"></exception>
         public ITokenAcquirer GetTokenAcquirer(AuthenticationOptions applicationAuthenticationOptions)
         {
-            throw new NotImplementedException();
+            if (applicationAuthenticationOptions is null)
+            {
+                throw new ArgumentNullException(nameof(applicationAuthenticationOptions));
+            }
+
+            // Compute the Azure region if the option is a MicrosoftAuthenticationOptions.
+            MicrosoftAuthenticationOptions? microsoftAuthenticationOptions = applicationAuthenticationOptions as MicrosoftAuthenticationOptions;
+            if (microsoftAuthenticationOptions == null)
+            {
+                microsoftAuthenticationOptions = new MicrosoftAuthenticationOptions
+                {
+                    AllowWebApiToBeAuthorizedByACL = applicationAuthenticationOptions.AllowWebApiToBeAuthorizedByACL,
+                    Audience = applicationAuthenticationOptions.Audience,
+                    Audiences = applicationAuthenticationOptions.Audiences,
+                    Authority = applicationAuthenticationOptions.Authority,
+                    ClientCredentials = applicationAuthenticationOptions.ClientCredentials,
+                    ClientId = applicationAuthenticationOptions.ClientId,
+                    SendX5C = applicationAuthenticationOptions.SendX5C,
+                    TokenDecryptionCredentials = applicationAuthenticationOptions.TokenDecryptionCredentials,
+                    EnablePiiLogging = applicationAuthenticationOptions.EnablePiiLogging,
+                };
+            }
+
+            // Compute the key
+            string key = GetKey(applicationAuthenticationOptions.Authority, applicationAuthenticationOptions.ClientId);
+            if (!authSchemes.TryGetValue(key, out ITokenAcquirer tokenAcquirer))
+            {
+                var optionsMonitor = ServiceProvider.GetRequiredService<IOptionsMonitor<MergedOptions>>();
+                var mergedOptions = optionsMonitor.Get(key);
+                MergedOptions.UpdateMergedOptionsFromMicrosoftAuthenticationOptions(microsoftAuthenticationOptions, mergedOptions);
+                tokenAcquirer = GetTokenAcquirer(key);
+            }
+            return tokenAcquirer;
+        }
+
+        /// <inheritdoc>
+        public ITokenAcquirer GetTokenAcquirer(string authenticationScheme)
+        {
+            if (!authSchemes.TryGetValue(authenticationScheme, out ITokenAcquirer acquirer))
+            {
+                var tokenAcquisition = ServiceProvider.GetRequiredService<ITokenAcquisition>();
+                acquirer = new TokenAcquirer(tokenAcquisition, authenticationScheme);
+                authSchemes.Add(authenticationScheme, acquirer);
+            }
+            return acquirer;
+        }
+
+        private static string GetKey(string? authority, string? clientId)
+        {
+            return $"{authority}{clientId}";
```

Comment:
not including the region any longer?

Reply:
I dont' think that a same application deployed instance could run in different regions.